plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/TestTimestamp.java

```diff
@@ -79,37 +84,42 @@ public void testTimestampBackedByInt64()
                     Optional.of(parquetSchema),
                     false);
 
-            Iterator<SqlTimestamp> expectedValues = timestamps.build().iterator();
-            try (ConnectorPageSource pageSource = StandardFileFormats.TRINO_PARQUET.createFileFormatReader(session, HDFS_ENVIRONMENT, tempFile.getFile(), columnNames, ImmutableList.of(TIMESTAMP_MILLIS))) {
-                // skip a page to exercise the decoder's skip() logic
-                Page firstPage = pageSource.getNextPage();
-
-                assertTrue(firstPage.getPositionCount() > 0, "Expected first page to have at least 1 row");
+            testReadingAs(timestampsMillis, TIMESTAMP_MILLIS, session, tempFile, columnNames);
+            testReadingAs(bigInts, BigintType.BIGINT, session, tempFile, columnNames);
+        }
+    }
 
-                for (int i = 0; i < firstPage.getPositionCount(); i++) {
-                    expectedValues.next();
-                }
+    private <T> void testReadingAs(ImmutableList.Builder<T> expectedValuesBuilder, Type type, ConnectorSession session, ParquetTester.TempFile tempFile, List<String> columnNames)
```

Comment:
Would be nice to introduce this method as a preparatory test refactor.

Reply:
expectedValuesBuilder -> expectedValues (don't pass builder here)

move this param as the last one

Reply:
I extracted the test refactoring into a separate commit